### PR TITLE
Create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by ryanlong1004**
Please do not close or merge.

Trying to see if we want to adopt it.

### CODE REVIEW

**Comment by ryanlong1004** [comment]
Results are integrated into the repository security tab.

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    # 📚 See https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsrun
+
+    #   If the Autobuild fails above, remove it and uncomment the following three lines. 
+    #   modify them (or add more) to build your code if your project, please refer to the EXAMPLE below for guidance.
+
+    # - run: |
+    #   echo "Run, Build Application using script"
+    #   ./location_of_script_within_repo/buildscript.sh
+
+    - name: Perform CodeQL Analysis
```
Comment (ope-adejumo) [inline .github/workflows/codeql-analysis.yml:71]: Does this type of Static Code analysis cover also Syntactic code analysis? Reading through this action I only see Semantic

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ develop ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ develop ]
+  schedule:
```
Comment (ope-adejumo) [inline .github/workflows/codeql-analysis.yml:20]: Why would we need a cron job to trigger this workflow? Would this be like for Reporting purposes to run at that time once a week?

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (ryanlong1004) [reply .github/workflows/codeql-analysis.yml:71]: Pylint takes care of that.  Should never reach this point before.

**Review by aerorahul — APPROVED** [review]
Good to have code quality analysis.
It would be good to see report of this analysis. 
Perhaps add a little blurb on how to read said analysis.

File: .github/workflows/codeql-analysis.yml (hunk 2 above)
Comment (ryanlong1004) [reply .github/workflows/codeql-analysis.yml:20]: I **believe** this has to do with database maintenance and perhaps syncing with Github.

**Pull request merged**